### DISCUSSION TITLE
Component prop types info for designers

### DISCUSSION THREAD

**Discussion by aweell** [Telefonica/mistica-design#1082] [Proposals]
Se plantea la necesidad de que los diseñadores puedan tener documentados los valores aceptados de un campo. **Esto aplica principalmente a elementos de texto**

Ej: La description de un callout solo permite una string, sin embargo actualmente en figma puedes formatear el texto tanto cambiando su estilo, como añadiendo formateos propios de markdown en el mismo.

Se plantean a priori varias posibilidades:

* Indicar en el campo de figma el tipo de propiedad que es.
* Indicar en las specs de componente el valor permitido para una propiedad.
* Indicar en la documentación del componente (Brand factory) una tabla de props donde se incluya esta información.

Algunas dudas:

* Debería darse una regla general (principalmente para los elementos de texto) donde siempre se entienda que es una string sin posibilidad de formateo a no ser que se indique lo contrario?
* Qué ocurre en los casos en los que por ejemplo se acepta un array de strings? ([Véase feedback screen](https://rebrand.ly/s4v8hos))
* Existe algún caso en el que pueda ser necesario documentar otra prop que no sea de texto?

**Comment by yceballost** [comment]
Dejo por algunas posibilidades:
- https://tinyurl.com/2poq5e9b (Haz hover en el radio de "Initials")
- https://tinyurl.com/2o3nwj6v

**Reply by aweell** [reply]
Nice! Debería ese tooltip llevar un link a una pequeña explicación de los diferentes data types / structures

[Link al ejemplo ->](https://mistica-web.vercel.app/playroom/#?code=N4Igxg9gJgpiBcIA8AFATjAbgSxgdwBUIIAbAZwAIB6APgB0A7JKbTRiisgFwE8SYAvMGDsOFAEYBDMAGsA5mggBXBlHgVIJCGjIA6KbIXLVAGlEdx22GgAyMAGZd1dEAEYADgA9OpbFBdmDGISVjBoAMKk2uqa2nosONaB5hTuEGTYXNgQDM4g9tieMP4ggcFo2HIAFk4UAAxlYnh%2BXFV5AEx17V4BKVUwlTV5rnV1mFW9QRQAvtOM9FNIAEIQ3u6SUCwMcgAaQq4AbNOpG1tyAJr7RwvBSADKXNIynOtggsCH0zfBFEgAShtsgBxRRKdwpYIMSQAW0ELkkmEkjzQLghYkRJCU7zkMC4DyRMAAFPDEciXABKCgAfipFBcdRccymPwoOXCVUk23eZFx%2BK4RJJSMkKJA5KZLO%2BLPuj1kL2k70%2BkpZvwBLAgSyUXC4OQoGKxAnpLhoAEloZIcUgqKrshqtTklVLrerNdqgnq4W4jWifoxvT8kERSFlwczlcFkTiuEIAzBPFwAMwmhiZbCSciWgixhM0cVhn6wMhgCruLI5A0gXjuGDqbgVbao0N51LpFNllz8RwNpvBWAkSQ8IT2NM83PdihZtCSIR%2BvPLVYnTbYbZEdxCAAcXxnypjcZsS%2BeVQw9nLNS47jI8CoVFgmBgWiraF00IgAC9sCQ%2B7ptHIqDAGABaABVO5rwgMAyCoAB1GBxCoAApBFJDuItsBLKgABEhQAfVrJQwC4JQMDIL1GzHCg-hgDYKGfDAKEkSxNQoKAhQoSsYEoTkoE4Lg0DwgiiK3KUqEzXd9wdbtLRWTxxLDUc81oLdLSdW1XRksR-kBZ07TdNN9RcdojWNSAmCtTSVPtRTlJdHV3XLeMjXCJRuAgaENCiNAlLM6yGDUy18VkcTPLVEFlHccTgEjPkBRARCyVKOkQAZUUKAEVKEvskBqQoYlSI4e47xgfCtyhWFy3sXjMmIkAtz7cQ73LSItBFLdbIi3lHn5YlwHcgIErYgBBCk5JZNkOS5AA1XT3kJd1KQEGhOHagkutiEUTF1KaxS3CAS2yBgyCEABtQSKGADbMWrPqeCrQb4v5OM8gIa6YAoW6ZkaJszvdPI2KWXr7tqFwnqrCg-sy6YPrzL6pp%2B57wn%2BrNHueih4fByGw2hi7YardCEYehLgZe3G0ZOzGsWxmAAFE8cBkBCYoamSdysQAF1hrEBTSMpdQVFgAoGGKdmkHZQqZEsTw-TAfpZGKIRIo66LLDQaxeqHcgXppCheYcJdii25nRs5HEhB5PEFa6pWVdFdmOBKj1LbCLsfjUlZlcd0jLRF2RxcCr2xdWSXpZkWW2rN5aXCXW8dBgVXhw12ltf5vWbdZBh2SN7kls6iOGCjnkhr9O3y0jsJ8%2Bq0i1ONXPS5jj2qD9n2Uj8mUZG%2BSTVgWS0Ek7hJRG4Ph3hEUNmigVo8jANMwEJEY6gAUgof8KE6bpPHJSZgn6QZaYnkgp5n8YF4oABWA4vDX0oUhYMh3D7Hg8nsfgJYv0M00qBhjX5aELwSt4GH5NaUgACsnJZHsDwSIf8-zbygY7Z%2BwQDDyFBKoRq0QKDy3DiAEu0cKRolpKtPQCCjC8yWJOVQaIYjuQIU8IhphRCzHmKIOc0km4EH6LCSaFRORcAoNgMgVc87YiztFLBZcxSBSriQXWco3jlmFMYEoalQ5RS6rFYUvVDTJVSgIBKSUKAADI9HZUUv1UkajmbwNCGgOWQiLaWILuYsQGQXzvAOAAFhTsEMgaAwAni1OeS8VBsBmhxHoFQ18%2BxkCqLoSA0IqDuCqBAbU-5XBH1cAAdjXK4gAnJko%2BR94xpP-K4qA7Qj5pKyV0VxR9XGSCpIUSR4gBBoHEMk3Q7RdCuD0YUPwAgACyDA8CeF6YA-q8Z7DoSWGuXpAAJTweBJBAgqeIHgrg8DnEglAEg9hZlrnsECcaShxmTO2Z4NcejJCaggAIew2gzRcD0QUKMRYdp6LwAINcaS6h6IAI7vKSluGgloTFCjQGpfWGN0HZxiqYkUc00ouFcC4fRhicpNiQMC5EnBsDOKEG444S4UzDnLL0u4RogUwrBSnJR5tBRxThdo-SSKDFGIceimFWKcXADxRQIyZZgC8oYIBHkaB%2BpgEgCoLgFE5BKD7GgL45KQWUr9NSjBqjYUpXhSADKyKWVooxWYscTiXHuJOgKoQAqhVhFFeKv%2BUqZXCg8WIAGKCrHAEJISOaC0h5kTcvtbhbEXUpTQTYlwq11EVmerdckABuE6HBsD2GygG9yGqGURpuhSU6caxAYH4kEfB%2BhSFQFjQ44IjrggJqTXDFNWirpVjBpSb1PqKC5sIvmyhugyB4TeGQMgJafXlrEJWwkyamqprrTAVGjbs0cFbWgdtTU9DzPnUuOQ-ayKDvjYmkd1ax21pcGxYm07S0-DnQuuIugwiKDQOuscm6eHbtHdocdB7nqM2Pc2ltuI21uUXbodwihny3u7EybNZ7f0XvEEW4DslyQesHYQpBUAXVCHdZ6rNJ6ODGW4KxXdz7tGQuimG%2BKr6M2ihg3mYdT60AvvTTAKNGHP3gYLVBziNgIB4Ao7JbNVG8M0f3XRhtjHm3MY7V2sV7EyDsc49m%2B9vGqyBoEwGzNTafWib-cu5M2xpNceVHJx9fHaOHpUzOr9eaIM6EvWga9OnZM8YMwpmtmq2LvuE2p7986LN6AAy5CAtnMMzF9AF9TkHoMnWmHB8FeZAVUH1aCv0UXbhUAkbrJUGY2EwA4amP%2BbcqBSU7teVgCxSgVgy2QBAh06YFRucmCeIAWYmBACPVo5X4CHVcPGA4a42ZAA)

Me pregunto...es equivalente entre plataformas?

**Reply by yceballost** [reply]
link not found D:

**Reply by yceballost** [reply]
Podriamos meter en cada tooltip lo que consideremos importante para cada prop, no? Me parece correcto poner ese tipo de elementos para ampliar información si fuese necesario

**Reply by AnaMontes11** [reply]
Totalmente agree. Tener la posibilidad de ampliar información de este tipo, puede ser muy útil

**Reply by aweell** [reply]
Apunto por aquí otra preocupación que surgió ayer sobre documentar este tipo de información en Figma y es que la relación de propiedades no necesariamente tiene que ser 1:1 entre la librería y la implementación, esto me empuja más a documentar esto o en los snippets de brand factory (teniendo en cuenta que a veces hay propiedades que no se muestran) o en storybook. Desde mi punto de vista este tipo de documentación debería ir en el último, el target no debería ser sólo diseñadores y centralizar todo en storybook posiblemente tenga un mejor mantenimiento.

**Comment by iciaparicio** [comment]
[surprised]     Iciar Aparicio Garcia reacted to your message:
________________________________
